### PR TITLE
Publish to artifactory

### PR DESCRIPTION
**Pull request by mhratson**
## Summary

- artifacts publications have been disabled since [2.5.11](https://github.com/linkedin/kafka-monitor/releases/tag/)
- which prevents apps that depend on KM to update to the latest versions
- PR adds Github Action to publish to JFrog Artifactory on release tag creation
- as well as Artifactory and Gradle integration

## Testing Done
1. [workflow ran successfully](https://github.com/linkedin/kafka-monitor/runs/8278325597?check_suite_focus=true#step:6:39) and 
2. publish went through — [test version 0.0.0 in JFROG](https://linkedin.jfrog.io/ui/packages/gav:%2F%2Fcom.linkedin.kmf:kafka-monitor/0.0.0?name=kafka-monitor&type=packages&activeTab=builds)

### CODE REVIEW

File: .github/workflows/tag.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      run: ./gradlew clean build javadoc --stacktrace --warning-mode all --no-daemon
+    - name: Branch tag
+      id: branch_tag
+      run: echo ::set-output name=RELEASE_TAG::${GITHUB_REF#refs/tags/}
+    - name: Publish to Jfrog
+      env:
+        JFROG_USER: ${{ secrets.JFROG_USER }}
+        JFROG_KEY: ${{ secrets.JFROG_KEY }}
+        RELEASE_TAG: ${{ steps.branch_tag.outputs.RELEASE_TAG }}
+      run: ./scripts/publishToJfrog.sh
```
Comment (lmr3796) [inline .github/workflows/tag.yml:34]: nit: Please add an EOF new line.

File: scripts/publishToJfrog.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    then
+        echo "publishing: tag $RELEASE_TAG looks like a semver"
+        git status
+        git describe --tags
+        ./gradlew printVersion
+        ./gradlew publishMyPublicationPublicationToLinkedInJfrogRepository
+    else
+        echo "not publishing: tag $RELEASE_TAG is NOT a valid semantic version (x.y.z)"
+    fi
+fi
```
Comment (lmr3796) [inline scripts/publishToJfrog.sh:21]: Ditto as above

File: scripts/publishToJfrog.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+else
+    echo "we are in $PWD and tag is $RELEASE_TAG"
+
+    if [[ $RELEASE_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]
+    then
+        echo "publishing: tag $RELEASE_TAG looks like a semver"
+        git status
+        git describe --tags
+        ./gradlew printVersion
+        ./gradlew publishMyPublicationPublicationToLinkedInJfrogRepository
```
Comment (lmr3796) [inline scripts/publishToJfrog.sh:17]: Is this goal implemented already?  I didn't see it in the PR

**Review by lmr3796 — CHANGES_REQUESTED** [review]

File: scripts/publishToJfrog.sh (hunk 3 above)
Comment (mhratson) [reply scripts/publishToJfrog.sh:17]: this goal/task is dynamic: gets generated from `build.gradle` publication declarations. 
IE [`LinkedInJfrog`](https://github.com/linkedin/kafka-monitor/pull/374/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R128) becames part of the goal name

**Review by lmr3796 — APPROVED** [review]

**Pull request merged**